Sources/SourceControl/GitRepository.swift

```diff
@@ -207,12 +210,14 @@ public struct GitRepositoryProvider: RepositoryProvider, Cancellable {
 
     public func isValidDirectory(_ directory: Basics.AbsolutePath) throws -> Bool {
         let result = try self.git.run(["-C", directory.pathString, "rev-parse", "--git-dir"])
-        return result == ".git" || result == "." || result == directory.pathString
```

Comment:
Actually, why are we even checking the resulting string at all? Doesn't `rev-parse` just fail if it isn't a git directory and that's all we care about in this function?

Reply:
Running it on a non-directory yields the following in the terminal:
fatal: not a git repository (or any of the parent directories): .git

Reply:
Hmm, so it returns:
```
       --git-dir
           Show $GIT_DIR if defined. Otherwise show the path to the .git directory. The path shown, when relative, is
           relative to the current working directory.

           If $GIT_DIR is not defined and the current directory is not detected to lie in a Git repository or work tree
           print a message to stderr and exit with nonzero status.
```

I'm not sure if this function is purposefully checking that $GIT_DIR isn't set or not?

I would be fine with you leaving this TBH, it's unrelated to the change you actually care about.